src/main/java/com/powsybl/openloadflow/network/VoltageControl.java

```diff
@@ -148,7 +148,7 @@ public boolean isHidden() {
             return true; // means all disabled
         } else {
             // we should normally have max 3 voltage controls (one of each type) because already merged
-            return voltageControls.get(0) != this;
+            return voltageControls.get(0) != this && voltageControls.get(0).getControlledBus() == this.getControlledBus();
```

Comment:
But a voltage control, with a higher priority but not the same controlled bus BUT with a controlled bus connected to same zero impedance network (this is the only case where voltageControls.get(0).getControlledBus() == this.getControlledBus()) also hide this voltage control?

Reply:
Yes, you are right.